streaming/src/main/scala/org/apache/spark/streaming/ui/AllBatchesTable.scala

```diff
@@ -156,40 +160,232 @@ private[ui] class ActiveBatchTable(
+      Some("Total time taken to handle a batch"), None)
+
+    assert(completedBatchTableHeaders.length == tooltips.length)
+
+    val headerRow: Seq[Node] = {
+      completedBatchTableHeaders.zip(tooltips).map { case (header, tooltip) =>
+        if (header == sortColumn) {
+          val headerLink = Unparsed(
+            parameterPath +
+              s"&$completedBatchTag.sort=${URLEncoder.encode(header, UTF_8.name())}" +
+              s"&$completedBatchTag.desc=${!desc}" +
+              s"&$completedBatchTag.pageSize=$pageSize" +
+              s"#$completedBatchTag")
+          val arrow = if (desc) "&#x25BE;" else "&#x25B4;" // UP or DOWN
+
+          if (tooltip.nonEmpty) {
+            <th>
+              <a href={headerLink}>
+                <span data-toggle="tooltip" title={tooltip.get}>
+                  {header}&nbsp;{Unparsed(arrow)}
+                </span>
+              </a>
+            </th>
+          } else {
+            <th>
+              <a href={headerLink}>
+                {header}&nbsp;{Unparsed(arrow)}
+              </a>
+            </th>
+          }
+        } else {
+          val headerLink = Unparsed(
+            parameterPath +
+              s"&$completedBatchTag.sort=${URLEncoder.encode(header, UTF_8.name())}" +
+              s"&$completedBatchTag.pageSize=$pageSize" +
+              s"#$completedBatchTag")
+
+          if(tooltip.nonEmpty) {
+            <th>
+              <a href={headerLink}>
+                <span data-toggle="tooltip" title={tooltip.get}>
+                  {header}
+                </span>
+              </a>
+            </th>
+          } else {
+            <th>
+              <a href={headerLink}>
+                {header}
+              </a>
+            </th>
+          }
+        }
+      }
+    }
+    <thead>
+      {headerRow}
+    </thead>
+  }
+
+  override def row(completedBatchRow: CompletedBatchTableRow): Seq[Node] = {
+    val batch = completedBatchRow.batchData
+    val batchTime = completedBatchRow.batchTime
+    val formattedBatchTime = UIUtils.formatBatchTime(batchTime, batchInterval)
+    val numRecords = completedBatchRow.numRecords
+    val schedulingDelay = completedBatchRow.schedulingDelay
+    val formattedSchedulingDelay = schedulingDelay.map(SparkUIUtils.formatDuration).getOrElse("-")
+    val processingTime = completedBatchRow.processingDelay
+    val formattedProcessingTime = processingTime.map(SparkUIUtils.formatDuration).getOrElse("-")
+    val batchTimeId = s"batch-$batchTime"
+    val totalDelay = completedBatchRow.totalDelay
     val formattedTotalDelay = totalDelay.map(SparkUIUtils.formatDuration).getOrElse("-")
 
-    baseRow(batch) ++ {
+    <tr>
+      <td id={batchTimeId} sorttable_customkey={batchTime.toString}>
+        <a href={s"batch?id=$batchTime"}>
+          {formattedBatchTime}
+        </a>
+      </td>
+      <td sorttable_customkey={numRecords.toString}>{numRecords.toString} records</td>
```

Comment:
Is it necessary to add `sortable_customkey`?

Reply:
No. Not required. Removed this.